File: api-signature/testsrc/test/esya/api/signature/settings/UGSettings.java
package test.esya.api.signature.settings;

import tr.gov.tubitak.uekae.esya.api.asn.x509.ECertificate;
import tr.gov.tubitak.uekae.esya.api.common.crypto.BaseSigner;
import tr.gov.tubitak.uekae.esya.api.crypto.alg.SignatureAlg;
import tr.gov.tubitak.uekae.esya.api.signature.Signable;
import tr.gov.tubitak.uekae.esya.api.signature.impl.SignableBytes;
import tr.gov.tubitak.uekae.esya.api.signature.util.PfxSigner;

/**
 * @author ayetgin
 */
public class UGSettings implements TestSettings
{
    private String pfxPath = "./api-signature/testresources/ahmet.yetgin_890111@ug.net.pfx";
    private String pfxPass = "890111";

    private PfxSigner signer;

    public UGSettings()
    {
        signer = new PfxSigner(SignatureAlg.RSA_SHA256, pfxPath, pfxPass.toCharArray());
    }

    public String getBaseDir()
    {
        return null;
    }

    public BaseSigner getSigner()
    {
        return signer;
    }

    public ECertificate getSignersCertificate()
    {
        return signer.getSignersCertificate();
    }

    public Signable getContent()
    {
        return new SignableBytes("test data".getBytes(), "data.txt", "text/plain");
    }
}

File: api-signature/csharp/test/src/test/esya/api/signature/settings/UGSettings.cs


using System;
using System.Text;
using tr.gov.tubitak.uekae.esya.api.asn.x509;
using tr.gov.tubitak.uekae.esya.api.common.crypto;
using tr.gov.tubitak.uekae.esya.api.crypto.alg;
using tr.gov.tubitak.uekae.esya.api.signature;
using tr.gov.tubitak.uekae.esya.api.signature.impl;
using tr.gov.tubitak.uekae.esya.api.signature.util;

namespace test.esya.api.signature.settings
{
    public class UGSettings : TestSettings
    {
        private String pfxPath = "E:\\ahmet\\prj\\MA3API\\MA3\\api-signature\\testresources\\ahmet.yetgin_890111@ug.net.pfx";
        private String pfxPass = "890111";

        private PfxSigner signer;

        public UGSettings()
        {
            signer = new PfxSigner(Algorithms.SIGNATURE_RSA_SHA256, pfxPath, pfxPass);
        }

        public String getBaseDir()
        {
            return null;
        }

        public BaseSigner getSigner()
        {
            return signer;
        }

        public ECertificate getSignersCertificate()
        {
            return signer.getSignersCertificate();
        }

        public Signable getContent()
        {
            return new SignableBytes(Encoding.ASCII.GetBytes("test data"), "data.txt", "text/plain");
        }
    }
}
